pkg/machine/shim/networking_unix.go

```diff
@@ -10,16 +10,22 @@ import (
 
 	"github.com/containers/podman/v5/pkg/machine"
 	"github.com/containers/podman/v5/pkg/machine/define"
+	"github.com/containers/podman/v5/pkg/machine/vmconfigs"
 	"github.com/sirupsen/logrus"
 )
 
-func setupMachineSockets(name string, dirs *define.MachineDirs) ([]string, string, machine.APIForwardingState, error) {
-	hostSocket, err := dirs.DataDir.AppendToNewVMFile("podman.sock", nil)
+func setupMachineSockets(mc *vmconfigs.MachineConfig) ([]string, string, machine.APIForwardingState, error) {
+	hostSocket, err := mc.APISocket()
 	if err != nil {
 		return nil, "", 0, err
 	}
 
-	linkSocketPath := filepath.Dir(dirs.DataDir.GetPath())
+	dataDir, err := mc.DataDir()
```

Comment:
is there a reason to not keep the dirs from the original signature or maybe just pass datadir in? it looks like the caller already has it ?

Reply:
Not a great one. Previously the machine config was not passed, so sig needed to change to bring over the API socket method, and so when I changed it I was thinking the config struct already has the data dir on it. Although since that method adds error handling  it does bloat the code here, so I'll restore passing it to delete that bit of code.